test/e2e/tests/4byte-directory.spec.js

```diff
@@ -0,0 +1,134 @@
+        const actionElement = await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Deposit',
+        });
+        assert.equal(await actionElement.getText(), 'DEPOSIT');
+      },
+    );
+  });
+
+  it('does not try to get contract method name from 4byte when the setting is off', async function () {
+    const smartContract = SMART_CONTRACTS.PIGGYBANK;
+    const fourBytePrefix = '0xd0e30db0';
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        smartContract,
+        title: this.test.title,
+        testSpecificMock: async (mockServer) => {
+          await mockServer
+            .forGet(`https://www.4byte.directory/api/v1/signatures/`)
+            .withQuery({ hex_signature: fourBytePrefix })
+            .thenCallback(() => {
+              return {
+                statusCode: 200,
+                json: {
+                  count: 1,
+                  next: null,
+                  previous: null,
+                  results: [
+                    {
+                      id: 1,
+                      created_at: null,
+                      text_signature: 'deposit()',
+                      hex_signature: null,
+                      bytes_signature: null,
+                    },
+                  ],
+                },
+              };
+            });
+        },
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        // goes to the settings screen
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Security & privacy', tag: 'div' });
+
+        // turns off 4Byte Directory contract method name resolution
+        await driver.clickElement(
+          '[data-testid="4byte-resolution-container"] .toggle-button',
+        );
+
+        // deploy contract
+        await openDapp(driver, contractAddress);
+
+        // wait for deployed contract, calls and confirms a contract method where ETH is sent
+        await driver.delay(largeDelayMs);
+        await driver.clickElement('#depositButton');
+
+        await driver.waitForSelector({
+          css: 'span',
+          text: 'Deposit initiated',
+        });
+
+        await driver.waitUntilXWindowHandles(3);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Notification);
+        const actionElement = await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Contract interaction',
+        });
+        await driver.delay(veryLargeDelayMs);
```

Comment:
So this is sort of a special case @pedronfigueiredo .  The default text that displays *whether the setting is on or off* is "CONTRACT INTERACTION"; if the setting is on, roughly a few seconds pass (to make the 4byte request) and "CONTRACT INTERACTION" gets changed to "DEPOSIT" right in the confirmation.  Essentially, the text gets changed on the fly once the 4byte request is done.

The delay was meant to check "let some time pass, then ensure that 'contract interaction' is still there".  We're checking that something *didn't* happen.

Let me know if you have another idea!

Reply:
Thanks for the clarification, David. That makes sense. One follow up question:

I'm not super familiar with this part of the code, but looking at the test, I see there are two different strings, `Contract interaction` and `CONTRACT INTERACTION`. Is the first one the default value and the second the value returned from 4byte after the call? If that's the case, this would presumably be enough to distinguish whether or not the call has happened.

Reply:
@darkwing was kind enough to jump on a call with me, and we had a good discussion. I understand the context now. Given that we need to wait for "things not to change" it makes sense to keep the delay in this instance.

To make things a bit more clear, though, can I ask you @vthomas13 to change the assertion below to 

```js
const actionElement = await driver.waitForSelector({
  css: '.confirm-page-container-summary__action__name',
  text: 'Contract interaction',
});
```

so it's evident it's the same assertion as above?

The fact that `getText()` returns text with a different capitalization is a bit confusing here.